src/systemctl/systemctl.c

```diff
@@ -8572,12 +8576,10 @@ static int halt_main(void) {
         if (r < 0)
                 return r;
 
-        if (arg_when > 0)
-#if ENABLE_LOGIND
-                return logind_schedule_shutdown();
-#else
-                log_error("Cannot schedule shutdown without logind support, proceeding with immediate shutdown.");
-#endif
+        /* Delayed shutdown requested, and was successful */
+        if (arg_when > 0 && logind_schedule_shutdown() == 0)
```

Comment:
If I understand the code correctly, `arg_when` seems always positive in `halt_main()`. So, this condition seems slightly redundant. Also, `logind_check_inhibitors()` just above always returns `0`.

Reply:
I believe when one calls `shutdown now`, `parse_shutdown_time_spec()` sets `arg_when` to 0, which then does check inhibitors, and scheduling should not be done in this case.

Reply:
Ah, I see. Thanks.